gammapy/cube/tests/test_edisp_map.py

```diff
@@ -121,3 +121,17 @@ def test_edisp_map_stacking():
     edmap_stack.stack(edmap2)
     assert_allclose(edmap_stack.edisp_map.data, edmap1.edisp_map.data)
     assert_allclose(edmap_stack.exposure_map.data, edmap1.exposure_map.data * 3)
+
+
+def test_sample_coord():
+    edisp_map = make_edisp_map_test()
+
+    coords = MapCoord(
+        {"lon": [0, 0] * u.deg, "lat": [0, 0.5] * u.deg, "energy": [1, 3] * u.TeV},
+        coordsys="CEL",
```

Comment:
If I remove it, then pytest gives me this error 
```======================================== FAILURES =========================================
____________________________________ test_sample_coord ____________________________________

    def test_sample_coord():
        edisp_map = make_edisp_map_test()
    
        coords = MapCoord(
            {"lon": [0, 0] * u.deg, "lat": [0, 0.5] * u.deg, "energy": [1, 3] * u.TeV}
        )
    
>       coords_corrected = edisp_map.sample_coord(map_coord=coords)

tests/test_edisp_map.py:133: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
edisp_map.py:352: in sample_coord
    "skycoord": map_coord.skycoord.reshape(-1, 1),
../maps/geom.py:878: in skycoord
    self.lon, self.lat, unit="deg", frame=coordsys_to_frame(self.coordsys)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

coordsys = None

    def coordsys_to_frame(coordsys):
        if coordsys in ["CEL", "C", "fk5", "fk4", "icrs"]:
            return "icrs"
        elif coordsys in ["GAL", "G", "galactic"]:
            return "galactic"
        else:
>           raise ValueError(f"Unrecognized coordinate system: {coordsys!r}")
E           ValueError: Unrecognized coordinate system: None

../maps/geom.py:178: ValueError
===================== 1 failed, 25 passed, 23 skipped in 3.52 seconds =====================
```

Hence, I think it is better to keep it.

Reply:
I see, OK to leave like this in this PR then.

@adonath - This is bad / error-prone:

https://github.com/gammapy/gammapy/blob/2504c567c9fd9aad31bf14dc07fde6c2e2dbfc79/gammapy/maps/geom.py#L801-L803

Allowing a "None" case for `coordsys` invites trouble, and creates if/else checks in many places.
It should be changed to internally only using SkyCoord, or to a separate `frame` attribute, and only allowing valid frames. Overall, the question is whether `MapCoord` always has to have sky coords, or if it can also be used to only hold other coords.

I think there's already issues open concerning MapCoord, probabaly a discussion to just be ignored for now ....